### PR TITLE
python3-coverage: update to 7.2.1

### PR DESCRIPTION
**Pull request by jason1987d**
#### Testing the changes
- I tested the changes in this PR: NO
#### Local build testing
- I built this PR locally for my native architecture, (ARCH-LIBC)

### CODE REVIEW

**Comment by jason1987d** [comment]
The result of running the tests.
```
=========================== short test summary info ============================
FAILED tests/test_concurrency.py::ConcurrencyTest::test_threads - AssertionEr...
FAILED tests/test_concurrency.py::ConcurrencyTest::test_threads_simple_code
FAILED tests/test_concurrency.py::ConcurrencyTest::test_gevent - assert '/bin...
FAILED tests/test_concurrency.py::ConcurrencyTest::test_eventlet - assert '/b...
FAILED tests/test_concurrency.py::ConcurrencyTest::test_greenlet - assert '/b...
FAILED tests/test_concurrency.py::ConcurrencyTest::test_bug_330 - assert '/bi...
FAILED tests/test_concurrency.py::ConcurrencyTest::test_gevent_simple_code - ...
FAILED tests/test_concurrency.py::ConcurrencyTest::test_greenlet_simple_code
FAILED tests/test_concurrency.py::SigtermTest::test_sigterm_still_runs - Asse...
FAILED tests/test_concurrency.py::ConcurrencyTest::test_eventlet_simple_code
FAILED tests/test_concurrency.py::ConcurrencyTest::test_threads_with_gevent
FAILED tests/test_plugins.py::PluginTest::test_local_files_are_importable - A...
FAILED tests/test_concurrency.py::SigtermTest::test_sigterm_saves_data[False]
FAILED tests/test_process.py::ProcessTest::test_erase_parallel - AssertionErr...
FAILED tests/test_process.py::ProcessTest::test_running_missing_file - Assert...
FAILED tests/test_process.py::FailUnderTest::test_report_99p9_is_not_ok - ass...
FAILED tests/test_process.py::ProcessTest::test_coverage_run_envvar_is_in_coveragerun
FAILED tests/test_process.py::YankedDirectoryTest::test_removing_directory_with_error
FAILED tests/test_process.py::ProcessTest::test_append_data_with_different_file
FAILED tests/test_summary.py::SummaryTest::test_report_skip_covered_no_branches
FAILED tests/test_process.py::EnvironmentTest::test_coverage_run_dashm_dir_with_init_is_like_python
FAILED tests/test_process.py::EnvironmentTest::test_coverage_run_dashm_equal_to_doubledashsource
FAILED tests/test_concurrency.py::MultiprocessingTest::test_multiprocessing_with_branching[fork]
FAILED tests/test_process.py::ProcessTest::test_tests_dir_is_importable - Ass...
FAILED tests/test_process.py::ProcessTest::test_append_can_create_a_data_file
FAILED tests/test_concurrency.py::MultiprocessingTest::test_multiprocessing_and_gevent[fork]
FAILED tests/test_process.py::ExcepthookTest::test_excepthook_throw - assert ...
FAILED tests/test_process.py::ProcessTest::test_missing_source_file - assert ...
FAILED tests/test_process.py::AliasedCommandTest::test_specific_alias_works
FAILED tests/test_process.py::ProcessTest::test_warnings_trace_function_changed_with_threads
FAILED tests/test_process.py::AliasedCommandTest::test_aliases_used_in_messages[coverage3]
FAILED tests/test_concurrency.py::MultiprocessingTest::test_multiprocessing_simple[fork]
FAILED tests/test_concurrency.py::MultiprocessingTest::test_multiprocessing_append[spawn]
FAILED tests/test_concurrency.py::MultiprocessingTest::test_multiprocessing_simple[spawn]
FAILED tests/test_summary.py::SummaryTest::test_report_just_one - ModuleNotFo...
FAILED tests/test_summary.py::SummaryTest::test_report_with_chdir - Assertion...
FAILED tests/test_process.py::ProcessTest::test_fullcoverage - assert 'Error ...
FAILED tests/test_concurrency.py::MultiprocessingTest::test_multiprocessing_with_branching[spawn]
FAILED tests/test_concurrency.py::MultiprocessingTest::test_multiprocessing_bootstrap_error_handling
FAILED tests/test_process.py::EnvironmentTest::test_coverage_run_dashm_superset_of_doubledashsource
FAILED tests/test_process.py::ProcessTest::test_combine_with_rc - AssertionEr...
FAILED tests/test_process.py::ProcessTest::test_combine_with_aliases - Assert...
FAILED tests/test_process.py::ProcessTest::test_warns_if_never_run - assert "...
FAILED tests/test_concurrency.py::MultiprocessingTest::test_multiprocessing_append[fork]
FAILED tests/test_process.py::EnvironmentTest::test_coverage_run_dashm_is_like_python_dashm_off_path
FAILED tests/test_concurrency.py::MultiprocessingTest::test_multiprocessing_and_gevent[spawn]
FAILED tests/test_process.py::ProcessTest::test_append_data - AssertionError:...
FAILED tests/test_process.py::EnvironmentTest::test_coverage_run_is_like_python
FAILED tests/test_process.py::ProcessTest::test_code_exits - AssertionError: ...
FAILED tests/test_process.py::ProcessTest::test_fork - AssertionError: assert...
FAILED tests/test_process.py::ProcessTest::test_warning_trace_function_changed
FAILED tests/test_process.py::EnvironmentTest::test_coverage_run_dashm_is_like_python_dashm
FAILED tests/test_process.py::EnvironmentTest::test_bug_862 - assert 'inside ...
FAILED tests/test_process.py::EnvironmentTest::test_coverage_run_dir_is_like_python_dir
FAILED tests/test_process.py::EnvironmentTest::test_bug_909 - AssertionError:...
FAILED tests/test_process.py::FailUnderTest::test_report_43_is_not_ok - asser...
FAILED tests/test_process.py::ProcessTest::test_timid - AssertionError: asser...
FAILED tests/test_process.py::EnvironmentTest::test_coverage_run_dashm_is_like_python_dashm_with__main__207
FAILED tests/test_concurrency.py::MultiprocessingTest::test_bug_890 - Asserti...
FAILED tests/test_process.py::FailUnderEmptyFilesTest::test_report - assert 1...
FAILED tests/test_process.py::AliasedCommandTest::test_major_version_works - ...
FAILED tests/test_process.py::FailUnderTest::test_report_43_is_ok - assert 12...
FAILED tests/test_process.py::ExcepthookTest::test_excepthook - assert 127 == 1
FAILED tests/test_process.py::YankedDirectoryTest::test_removing_directory - ...
FAILED tests/test_python.py::GetZipBytesTest::test_get_encoded_zip_files[utf-8]
FAILED tests/test_concurrency.py::SigtermTest::test_sigterm_saves_data[True]
FAILED tests/test_process.py::ProcessStartupTest::test_subprocess_with_pth_files_and_parallel
FAILED tests/test_python.py::GetZipBytesTest::test_get_encoded_zip_files[gb2312]
FAILED tests/test_python.py::GetZipBytesTest::test_get_encoded_zip_files[hebrew]
FAILED tests/test_filereporter.py::FileReporterTest::test_zipfile - ModuleNot...
FAILED tests/test_python.py::GetZipBytesTest::test_get_encoded_zip_files[shift_jis]
FAILED tests/test_python.py::GetZipBytesTest::test_get_encoded_zip_files[cp1252]
FAILED tests/test_process.py::EnvironmentTest::test_coverage_run_dashm_dir_no_init_is_like_python
FAILED tests/test_oddball.py::MockingProtectionTest::test_os_path_exists - As...
FAILED tests/test_summary.py::SummaryTest::test_report_wildcard - AssertionEr...
FAILED tests/test_process.py::ProcessTest::test_save_on_exit - AssertionError...
FAILED tests/test_summary.py::SummaryTest::test_report_including - ModuleNotF...
FAILED tests/test_process.py::ExcepthookTest::test_excepthook_exit - assert 1...
FAILED tests/test_summary.py::SummaryTest::test_omit_files_here - AssertionEr...
FAILED tests/test_testing.py::test_xdist_sys_path_nuttiness_is_fixed - Assert...
FAILED tests/test_summary.py::SummaryTest::test_report - ModuleNotFoundError:...
FAILED tests/test_process.py::ProcessTest::test_code_throws - assert '/bin/sh...
FAILED tests/test_testing.py::CoverageTestTest::test_sub_python_is_this_python
FAILED tests/test_process.py::EnvironmentTest::test_coverage_run_script_imports_doubledashsource
FAILED tests/test_process.py::ProcessTest::test_code_exits_no_arg - Assertion...
FAILED tests/test_process.py::ProcessTest::test_lang_c - AssertionError: asse...
FAILED tests/test_process.py::EnvironmentTest::test_coverage_zip_is_like_python
FAILED tests/test_process.py::AliasedCommandTest::test_aliases_used_in_messages[coverage]
FAILED tests/test_process.py::FailUnderTest::test_report_42p86_is_not_ok - as...
FAILED tests/test_process.py::EnvironmentTest::test_coverage_run_far_away_is_like_python
FAILED tests/test_summary.py::SummaryTest::test_run_omit_vs_report_omit - Ass...
FAILED tests/test_process.py::AliasedCommandTest::test_aliases_used_in_messages[coverage-3.11]
FAILED tests/test_process.py::FailUnderNoFilesTest::test_report - AssertionEr...
FAILED tests/test_summary.py::SummaryTest::test_report_omitting - ModuleNotFo...
FAILED tests/test_venv.py::VirtualenvTest::test_third_party_venv_isnt_measured[coverage]
FAILED tests/test_venv.py::VirtualenvTest::test_third_party_venv_isnt_measured[python -m coverage]
ERROR tests/test_numbits.py
96 failed, 1211 passed, 10 skipped, 1 error in 11.84s
=> ERROR: python3-coverage-7.2.1_5: do_check: 'PYTHONPATH="$(cd build/lib* && pwd)" ${make_check_pre} python3 -m pytest ${testjobs} ${make_check_args} ${make_check_target}' exited with 1
=> ERROR:   in do_check() at common/build-style/python3-module.sh:15
```

**Comment by jason1987d** [comment]
Any suggestions though on fixing the tests? make_check=no was there before on the 5.x version too.

**Comment by icp1994** [comment]
You might need to replicate the upstream test workflow for the tests to run properly https://github.com/nedbat/coveragepy/blob/7.2.1/tox.ini#L35-L48

File: srcpkgs/python3-coverage/template [hunk 1]
```diff
@@ -1,17 +1,16 @@
...
 build_style=python3-module
 hostmakedepends="python3-setuptools"
 makedepends="python3-devel"
 depends="python3-setuptools"
 checkdepends="python3-pytest-xdist python3-flaky python3-unittest-mixins"
 short_desc="Code coverage testing for Python3"
-maintainer="Orphaned <orphan@voidlinux.org>"
+maintainer="Jason Manley <jason@jasondavid.tv>"
 license="Apache-2.0"
 homepage="https://bitbucket.org/ned/coveragepy"
```
Comment (classabbyamp) [inline srcpkgs/python3-coverage/template:13]:
```suggestion
homepage="https://github.com/nedbat/coveragepy"
changelog="https://raw.githubusercontent.com/nedbat/coveragepy/master/CHANGES.rst"
```

File: srcpkgs/python3-coverage/template [hunk 2]
```diff
@@ -1,17 +1,16 @@
 # Template file for 'python3-coverage'
 pkgname=python3-coverage
-version=5.0.3
-revision=5
+version=7.2.1
```
Comment (classabbyamp) [inline srcpkgs/python3-coverage/template:3]: .2 is available

File: srcpkgs/python3-coverage/patches/no-versioned-coverage.patch [hunk 3]
```diff
@@ -1,16 +0,0 @@
---- a/setup.py
```
Comment (classabbyamp) [inline srcpkgs/python3-coverage/patches/no-versioned-coverage.patch:1]: this patch is still needed: https://github.com/void-linux/void-packages/actions/runs/4400142292/jobs/7705163969?pr=42735#step:9:20

Comment (tornaria) [reply srcpkgs/python3-coverage/patches/no-versioned-coverage.patch:1]: The patch could be replaced with the equivalent:
```
post_install() {
       # remove versioned scripts (keep only /usr/bin/coverage)
       rm ${DESTDIR}/usr/bin/coverage3
       rm ${DESTDIR}/usr/bin/coverage-${py3_ver}
}
```

**Pull request closed without merging**